django/db/models/aggregates.py

```diff
@@ -85,6 +85,10 @@ def resolve_expression(
             return c
         if hasattr(default, "resolve_expression"):
             default = default.resolve_expression(query, allow_joins, reuse, summarize)
+            if default._output_field_or_none is None:
+                default.output_field = c._output_field_or_none
+        else:
+            default = Value(default, c._output_field_or_none)
```

Comment:
Isn't that something that we could do in advance? :thinking:

Reply:
Do you mean in a separate commit? It use to be in a separate commit that focused on `contrib.postgres.aggregates` and its associated tests but I figured I'd merge them as I was struggling to write docs for two commits since they are closely related. Would you prefer if I have it another shot?

Reply:
I think we can treat this as a bugfix, see #16340.

Reply:
That makes sense, thanks!